tests/rptest/tests/partition_reassignments_test.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import random
+import json
+import re
+
+from ducktape.utils.util import wait_until
+from ducktape.tests.test import TestLoggerMaker
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+
+from rptest.clients.types import TopicSpec
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+
+
+def get_topics_and_partitions(reassignments: dict):
+    topic_names = []
+    partition_idxs = []
+    for reassignment in reassignments["partitions"]:
+        topic_names.append(reassignment["topic"])
+        partition_idxs.append(reassignment["partition"])
+
+    return topic_names, partition_idxs
+
+
+def execute_reassign_partitions(kafka_tools: KafkaCliTools,
```

Comment:
This works, FWIW I think it does make sense for some of this code to stay in `KafkaCLITools` like calling `reassingn_partitions` and parsing the `output`. Some commands parse the output into well defined structs / lists /dicts and returns that. Then the test can more easily work with the result sets.